.github/CONTRIBUTING.md

```diff
@@ -150,18 +159,73 @@ about how it works.
+Dredd's documentation is written in [Markdown][] using [Sphinx][]. [ReadTheDocs][] is used to build and publish the documentation:
 
 - [https://dredd.readthedocs.io](https://dredd.readthedocs.io) - preferred long URL
 - [http://dredd.rtfd.org](http://dredd.rtfd.org) - preferred short URL
 
-Source of the documentation can be found in the [docs][] directory. To contribute to Dredd's documentation, you will need to follow the [MkDocs installation instructions](http://www.mkdocs.org/#installation). Once installed, you may use following commands:
+Source of the documentation can be found in the [docs][] directory. To render Dredd's documentation on your computer, you need Python 3 and Node.js installed.
+
+#### Installation and Development
+
+1.  Make sure `node` is an executable and `npm install` has been done for the Dredd directory. Extensions to the docs are written in Node.js and Sphinx needs to have a way to execute them.
+2.  [Get Python 3](https://www.python.org/downloads/). On macOS, run `brew install python3`.
+3.  Create a [virtual environment](https://docs.python.org/3/library/venv.html) and activate it:
+
+    ```sh
+    python3 -m venv ./env
+    . ./env/bin/activate
+    ```
+
+4.  Install dependencies for the docs: `pip install -r docs/requirements.txt`
+
+Once installed, you may use following commands:
 
 - `npm run docs:build` - Builds the documentation
-- `npm run docs:serve` - Runs live preview of the documentation
+- `npm run docs:serve` - Runs live preview of the documentation on `http://127.0.0.1:8000`
+
+#### Installation on ReadTheDocs
+
+The final documentation gets deployed on the [ReadTheDocs][]. The service, however, does not support Node.js. Therefore on ReadTheDocs, the `conf.py` configuration file for Sphinx runs `docs/install-node.sh`, which installs Node.js locally, using [nvm][].
+
+#### ToC and Markdown
+
+Traditionally, Sphinx only supported the [reStructuredText][] format. Thanks to the [recommonmark][] project it's possible to use also [Markdown][], _almost_ as a format native to Sphinx. Dredd's docs are using the [AutoStructify][] extension to be able to specify _toctree_ and other stuff specific to reStructuredText. The ToC is generated from the _Contents_ section in the `docs/index.md` file.
+
+[recommonmark]: https://github.com/rtfd/recommonmark
+[AutoStructify]: https://recommonmark.readthedocs.io/en/latest/auto_structify.html
+
+#### Node.js Extensions
+
+There are some extensions hooked into the build process of [Sphinx][], modifying how the documents are processed. They're written in Node.js, because:
+
+- It's better to have them in the same language as Dredd.
+- This way they're able to import source files (e.g. `src/options.coffee`).
+
+<!-- To display the Hercule syntax, escaping is needed, because this document itself gets processed by Hercule. But escaping by backslashes doesn't work inside Markdown code blocks, so manual <code> element is added. -->
```

Comment:
Nope, it refers to the upcoming paragraph:

```
<!-- To display the Hercule syntax, escaping is needed, because this document itself
gets processed by Hercule. But escaping by backslashes doesn't work inside
Markdown code blocks, so manual <code> element is added. -->

By default, [Hercule][] is attached as an extension, which means you can
use the <code class="docutils literal"><span class="pre">:\[Title](link.md)</span></code>
syntax for including other Markdown files. All other extensions are custom and
are automatically loaded from the `docs/_extensions` directory.
```